File: recursion.py
def sum(x):
    if x == 0:
        return 0
    else:
        return x + sum(x-1)

def factorial(x):
    if x == 0:
        return 1
    else:
        return x*factorial(x-1)

def fibonachi(x):
    if x == 0:
        return 0
    elif x == 1:
        return 1
    else:
        return fibonachi(x-1) + fibonachi (x-2)

number = int(input('Input number: '))
print('sum(x) = ', sum(number))  
print('factorial(x) = ', factorial(number))
print('fibonachi(x) = ', fibonachi(number))

File: recursion.rb
def Factorial(x)
    if x == 0
        return 1
    else
        return x*Factorial(x-1)
    end
end

def Sum(x)
    if x == 0
        return 0
    else
        return x+Sum(x-1)
    end
end

def Fibonachi(x)
    if x == 0
        return 0
    elsif x == 1
        return 1
    else
        return Fibonachi(x-1) + Fibonachi(x-2)
    end
end

printf("Input number: ")
number = gets.encode('UTF-8').chomp.to_i
puts "Sum(x) = #{Sum(number)}"
puts "Factorial(x) = #{Factorial(number)}"
puts "Fibonachi(x) = #{Fibonachi(number)}"